km/km_filesys.c

```diff
@@ -2660,13 +2650,13 @@ static int km_fs_recover_open_file(char* ptr, size_t length)
 
    int fd = open(name, nt_file->flags, 0);
```

Comment:
codeql has decided we read a filename from user input here and then opening the file.

It seems like codeql has not gone deep enough to discover what km has opened and is reading the filename from.

We are reading from a snapshot file.  I guess codeql thinks that users manually generate snapshot files?

I guess we need to investigate what codeql is having fits about.

Reply:
We can squelch this, tell it it's false positive